junit-runtime/src/main/scala/org/junit/Assert.scala

```diff
@@ -0,0 +1,349 @@
+/*
+ * Ported from https://github.com/junit-team/junit
+ */
+package org.junit
+
+import org.junit.internal.InexactComparisonCriteria
+import org.junit.internal.ExactComparisonCriteria
+import org.hamcrest.Matcher
+import org.hamcrest.MatcherAssert
+
+object Assert {
+  def assertTrue(message: String, condition: Boolean): Unit =
+    if (!condition) fail(message)
+
+  def assertTrue(condition: Boolean): Unit =
+    assertTrue(null, condition)
+
+  def assertFalse(message: String, condition: Boolean): Unit =
+    assertTrue(message, !condition)
+
+  def assertFalse(condition: Boolean): Unit =
+    assertFalse(null, condition)
+
+  def fail(message: String): Unit = {
+    if (message eq null)
+      throw new AssertionError()
+    throw new AssertionError(message)
```

Comment:
I'm pretty sure this is equivalent to just always throw `new AssertionError(message)`, even when `message eq null`.